pkg/kn/commands/eventtype/list.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventtype
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/commands/flags"
+	hprinters "knative.dev/client/pkg/printers"
+)
+
+var listExample = `
+  # List all eventtypes
+  kn eventtype list
+
+  # List all eventtypes in JSON output format
+  kn eventtype list -o json`
+
+// NewEventtypeListCommand represents command to list all eventtypes
+func NewEventtypeListCommand(p *commands.KnParams) *cobra.Command {
+	listFlags := flags.NewListPrintFlags(ListHandlers)
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List eventtypes",
+		Aliases: []string{"ls"},
+		Example: listExample,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			eventingV1Beta1Client, err := p.NewEventingV1beta1Client(namespace)
+			if err != nil {
+				return err
+			}
+
+			eventTypeList, err := eventingV1Beta1Client.ListEventtypes(cmd.Context())
+			if err != nil {
+				return err
+			}
+			if !listFlags.GenericPrintFlags.OutputFlagSpecified() && len(eventTypeList.Items) == 0 {
+				fmt.Fprintf(cmd.OutOrStdout(), "No eventtypes found.\n")
+				return nil
+			}
+
+			// empty namespace indicates all-namespaces flag is specified
+			if namespace == "" {
+				listFlags.EnsureWithNamespace()
+			}
+
+			err = listFlags.Print(eventTypeList, cmd.OutOrStdout())
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	commands.AddNamespaceFlags(cmd.Flags(), true)
+	listFlags.AddFlags(cmd)
+	return cmd
+}
+
+// ListHandlers handles printing human readable table for `kn eventtype list` command's output
+func ListHandlers(h hprinters.PrintHandler) {
+	eventTypeColumnDefinitions := []metav1beta1.TableColumnDefinition{
```

Comment:
The list columns might be a bit tricky. At a first glance it seems to me that the definition is quiet dense. We've had similar issues with other listings. I'll try to find some references. I'd suggest to at least remove `Conditions`. And try to shortedn `Type` to `T` maybe, there's certainly a functions for it already.